src/engine/controls/bpmcontrol.cpp

```diff
@@ -937,64 +968,60 @@ void BpmControl::trackLoaded(TrackPointer pNewTrack) {
     mixxx::BeatsPointer pBeats;
     if (pNewTrack) {
         pBeats = pNewTrack->getBeats();
+        m_sampleRate = static_cast<mixxx::audio::SampleRate>(pNewTrack->getSampleRate());
```

Comment:
It might happen that a control update is received in between. Please revert.

Reply:
@uklotzde suggested creating a new method in `Track` to return both beats and sample rate as a single struct with locking. Should I do something like that?

Reply:
No, pBeats has already a sample rate and it is already received atomic. So there is no need to hack around that. 
Otherwise you have still the ongoing issue that both sample rates can differ later.